File: backend/app/dm/dashboard/schemas.py
"""系統儀表板（US7 / DM00）schema。回應對齊 spec_us7 FR-002 / FR-003。"""

from datetime import datetime

from pydantic import BaseModel


class CategoryStat(BaseModel):
    """單一分類之已發布目前版本統計。"""

    category_code: str
    category_name: str
    count: int


class DashboardStats(BaseModel):
    """各類型文件總數（4 內建分類 + 總計）。"""

    items: list[CategoryStat]
    total: int


class AnnouncementItem(BaseModel):
    """最新更新公告列（近 30 天已發布之版本）。"""

    doc_id: str
    doc_name: str
    category_code: str
    version_no: str
    change_summary: str | None
    published_date: datetime
    author_name: str | None
    kind: str  # NEW（新增首版）/ NEW_VERSION（新版本）

File: frontend/src/dm/dashboard/schemas.ts
/** 系統儀表板（US7 / DM00）型別（對齊後端 app/dm/dashboard/schemas.py）。 */

export interface CategoryStat {
  category_code: string
  category_name: string
  count: number
}

export interface DashboardStats {
  items: CategoryStat[]
  total: number
}

export interface AnnouncementItem {
  doc_id: string
  doc_name: string
  category_code: string
  version_no: string
  change_summary: string | null
  published_date: string
  author_name: string | null
  kind: string // NEW（新增）/ NEW_VERSION（新版本）
}

/** 類型 badge 顯示名。 */
export const KIND_LABELS: Record<string, string> = {
  NEW: "新增",
  NEW_VERSION: "新版本",
}
